src/number.rs

```diff
@@ -294,7 +294,8 @@ impl Display for Number {
         match self.n {
             N::PosInt(u) => Display::fmt(&u, formatter),
             N::NegInt(i) => Display::fmt(&i, formatter),
-            N::Float(f) => Display::fmt(&f, formatter),
+            // Preserve `.0` on integral values, which Display hides
+            N::Float(f) => Debug::fmt(&f, formatter),
```

Comment:
Actually we should just serialize using `ryu` and `itoa`, like the Serializer does, or maybe even go through the Serializer itself. `std::fmt` produces a worse representation for floats, like `120000000000000000000000000000000000000000` instead of `1.2e41`.